OpenRA.Game/Activities/Activity.cs

```diff
@@ -200,8 +200,29 @@ public Activity TickOuter(Actor self)
 		/// </summary>
 		protected virtual void OnLastRun(Actor self) { }
 
+		/// <summary>
+		/// Used only by Actor.Dispose() to make sure the activity finishes properly on death/disposal of actor
+		/// </summary>
+		public virtual void ForceCancel(Actor self)
```

Comment:
I think it would be better to call this `OnActorDispose` instead, which does nothing by default.

The individual activity subclasses can then choose to override this with their own cleanup logic, avoiding any side-effects that might otherwise happen by blindly calling `OnLastRun` without knowing what the activities might be doing inside there.

Reply:
I can do that, but I'm not convinced that OnLastRun being skipped on actor death/disposal is intentional.
Pinging @obrakmann just in case he has time to chime in on this.

Reply:
Tbh, I didn't quite think of modifying third-party state in the OnLastRun method, so the thought hasn't even crossed my mind. But I do agree with Paul here that unconditionally running OnLastRun might not be such a hot idea, and it should be up the activities themselves to decide what needs cleaning up after an actor gets disposed of.